### PR TITLE
ART-2823: update doozer release:gen-payload to sync split images

### PR DESCRIPTION
**Pull request by vfreex**
Be aware of the new `payload_name` field in image meta.
When syncing images, arch-specific image with explicit payload_name
serves as an override (https://issues.redhat.com/browse/ART-2823).

This PR also fixes a type error when `images_missing_builds` is not
empty, and changes the stand-in image from `cli` to `pod`.

### CODE REVIEW

File: doozerlib/cli/release_gen_payload.py [hunk 1]
```diff
@@ -324,13 +319,17 @@ def _get_mirror_sources(self, latest_builds, mismatched_siblings):
                     if record.private:  # exclude embargoed images from the ocp[-arch] imagestreams
                         yellow_print(f"Omitting embargoed image {pullspecs[-1]}")
                     else:
-                        self.runtime.logger.info(f"Adding {arch} image {pullspecs[-1]} to the public mirroring list with imagestream tag {tag_name}...")
-                        mirroring.setdefault((arch, False), {})[tag_name] = mirroring_value
+                        mirroring_list = mirroring.setdefault((arch, False), {})
+                        if tag_name not in mirroring_list or payload_name:  # arch-specific image with explicit payload_name serves as an override (https://issues.redhat.com/browse/ART-2823)
```
Comment (sosiouxme) [inline doozerlib/cli/release_gen_payload.py:323]: it took me a minute to puzzle out how this is an override. might this be clearer as:
```suggestion
                        if tag_name not in mirroring_list or payload_name:
                            # if multiple images in arch have the same tag, only explicit payload_name overwrites (https://issues.redhat.com/browse/ART-2823
```

**Review by sosiouxme — APPROVED** [review]
Unit tests or functional tests would be very welcome here, this is one of those edge cases that will be easy to forget about. But I recognize retrofitting this section to enable either may be too onerous.

File: doozerlib/cli/release_gen_payload.py [hunk 2]
```diff
@@ -297,6 +292,9 @@ def _get_mirror_sources(self, latest_builds, mismatched_siblings):
             if image.distgit_key in mismatched_siblings:
                 error = "Siblings built from different commits"
             else:
+                # The tag that will be used in the imagestreams
+                payload_name = image.config.get("payload_name")
+                tag_name = payload_name or image.image_name_short[4:] if image.image_name_short.startswith("ose-") else image.image_name_short
```
[the remark this answers is not in the document]
Comment (vfreex) [reply doozerlib/cli/release_gen_payload.py]: Review suggestions applied.
